contrib/install.sh

```diff
@@ -0,0 +1,1298 @@
+    shift
+    ;;
+  -h | --help)
+    show_usage
+    exit 0
+    ;;
+  --)
+    shift
+    break
+    ;;
+  *)
+    echo "Invalid option: $1" >&2
+    show_usage
+    exit 1
+    ;;
+  esac
+done
+
+# func: validate_network
+#
+# This function validates the provided network. If an invalid network is provided, this function
+# will show an error and abort the execution with exit 1
+#
+# Valid networks are:
+#   - bitcoin
+#   - signet
+#   - testnet
+#   - regtest
+validate_network() {
+  valid_networks=("bitcoin" "signet" "testnet" "regtest")
+  if [[ ! " ${valid_networks[@]} " =~ " ${network} " ]]; then
+    echo "❌ Invalid network '$network'. Valid options are: ${valid_networks[*]}"
+    exit 1
+  fi
+}
+
+# func: apt_install
+#
+# Check for missing packages (currently, only debian-like)
+# TODO: verify for fedora, arch and other systems
+apt_install() {
+  local missing=()
+
+  for package in "$@"; do
+    if ! dpkg -l | grep -q "^ii.*$package"; then
+      missing+=("$package")
+    fi
+  done
+
+  # Install curl first
+  sudo apt-get install -y -q curl
+
+  # Install missing packages if any
+  if [ ${#missing[@]} -gt 0 ]; then
+    echo "🐧 The following packages are missing: ${missing[*]}"
+    echo "🐧 Updating package lists..."
+    sudo apt update
+    echo "🐧 Installing missing packages..."
+    sudo apt-get install -y -q "${missing[@]}"
+  else
+    echo "🐧 All packages installed: skip"
+  fi
+}
+
+# func: install_rustup
+#
+# Install rustup with the recommended procedure (https://sh.rustup.rs)
+install_rustup() {
+  echo "🦀 Checking for rustup."
+  haveRust=$(which rustup)
+  if [ -z "$haveRust" ]; then
+    echo "🦀 Rustup not found. Installing..."
+    export RUSTUP_HOME=/home/$me/.rustup
+    export CARGO_HOME=/home/$me/.cargo
+    export PATH=/home/$me/.cargo/bin:$PATH
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y --no-modify-path
+
+    # If this script run with sudo, it will will make '.cargo'
+    # dir be owned by root. Change recursively this so the user will be the owner
+    chown $me:$me -R /home/$me/.cargo
+
+    # Load it on profile
+    echo "source /home/$me/.cargo/env" >>/home/$me/.profile
+    source /home/$me/.cargo/env
+  else
+    echo "🦀 Rustup found in $haveRust. Skip download..."
+  fi
+
+  isUpdated=$(/home/$me/.cargo/bin/rustup check | grep -B 1 "Up to date")
+  if [ -z "$isUpdated" ]; then
+    echo "🦀 Updating rust..."
+    /home/$me/.cargo/bin/rustup update
+  else
+    echo "🦀 Skip rust update..."
+  fi
+}
+
+# func: download_floresta
+#
+# Download floresta from releases and check integrity
+# TODO: add signature verification when possible
+download_floresta() {
+  echo "🌳 Download floresta $defaultTag..."
+  curl -L -o $tarDest --retry 5 --retry-delay 10 "$tarSrc"
+
+  sha256res=$(sha256sum $tarDest | awk -F" " '{ print $1 }')
+  sha256exp="733b1e2bcecfdf5ab552b81db428b125e6f8154cccfda69b5e746da7a4a0a322"
+  if [ ! "$sha256res" == "$sha256exp" ]; then
```

Comment:
Something that could be done in a follow-up: even on error, we should call the clean-up methods. Even better would be to remove what was already copied over, so we don't leave garbage back from an unsuccessful install.